c10/util/typeid.h

```diff
@@ -411,7 +427,7 @@ class C10_API TypeMeta final {
   }
 
   template <class T>
-  static C10_TYPENAME_CONSTEXPR c10::string_view TypeName() noexcept {
+  static c10::string_view TypeName() noexcept {
```

Comment:
Err, why did constexpr get removed here?

Reply:
Because it's a conditional macro, e.g. it's [not actually constexpr for GCC < 9](https://github.com/pytorch/pytorch/blob/master/c10/util/TypeIndex.h#L16-L55). Since no fundamental optimizations can rely on it - for instance, that's why `scalarTypeMetadatas` needs to be a static local rather than top level - I thought it was better to forego the extra obfuscation, build modality and so on and wait until we can provide total coverage.